bsms.mediawiki

```diff
@@ -0,0 +1,153 @@
+* The Signer generates a key record by prompting the user for a multisig derivation path and retrieves the XPUB at that derivation path. Optionally, the Signer can choose a path on behalf of the user. If the Signer chooses the path, it should try to avoid reusing XPUBs for different wallets.
+* The first line in the record must be the <tt>TOKEN</tt>. The second line must be the <tt>KEY</tt>. The <tt>KEY</tt> is an XPUB plus its key origin information, written in the descriptor-defined format, i.e.: <tt>[{master key fingerprint}/{derivation path}]{XPUB}</tt>. The third line must be a <tt>SIG</tt>, whereas <tt>SIG</tt> is the signature generated by using the private key associated with the XPUB to sign the first two lines.  The signature should follow [https://github.com/bitcoin/bips/blob/master/bip-0322.mediawiki BIP-0322], legacy format accepted. Finally, the Signer encrypts the entire record with <tt>ENCRYPTION_KEY</tt>.
+
+====Round 2====
+
+=====Coordinator=====
+
+* The Coordinator gathers key records from all participating Signers. Abort the setup if the wallet setup session has expired.
+* For each key record, the Coordinator decrypts it using <tt>ENCRYPTION_KEY</tt>. The Coordinator verifies that the included <tt>SIG</tt> is valid given the <tt>KEY</tt>.
+* If all key records look good, the Coordinator fills in all necessary information to generate a descriptor record, which is simply the descriptor string plus a <tt>CHECKSUM</tt>, all in one line. The <tt>CHECKSUM</tt> has [https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md#checksums BECH32 encoding].
+* The Coordinator encrypts this descriptor record with <tt>ENCRYPTION_KEY</tt>.
+* The Coordinator sends the encrypted descriptor record to all participating Signers.
+
+=====Signer=====
+
+* The Signer imports the descriptor record, decrypts it using the <tt>ENCRYPTION_KEY</tt> derived from the open session.
+* The Signer calculates and verifies the descriptor’s <tt>CHECKSUM</tt>. Abort the setup if the <tt>CHECKSUM</tt> is incorrect.
+* The Signer checks whether one of the <tt>KEY</tt>s in the descriptor belongs to it, using path and fingerprint information included in the descriptor. The check must perform an exact match on the <tt>KEY</tt>s, and not using shortcuts such as matching fingerprints (which is trivial to spoof). Abort the setup if it doesn’t detect its own <tt>KEY</tt>.
+* For confirmation, the Signer must display to the user the <tt>CHECKSUM</tt>, plus other configurations, such as <tt>M</tt> and <tt>N</tt>. The total number of Signers, <tt>N</tt>, is important to prevent a <tt>KEY</tt> insertion attack. All participating Signers should be able to display the same confirmation.
+* If all checks pass, the Signer persists the descriptor record in its storage.
+* The Signer can choose to further restrict post-XPUB derivation paths, such as to those defined in [https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki BIP-0044].
```

Comment:
Yes. It should abort.

This is related to our discussion earlier about whether the Coordinator should generate different descriptor strings for change/receive (and potentially other indices in the future).

The downside of that approach is that user will now have to validate _two strings_ instead of one, doubling the effort of what's already a fairly involving process. So instead of having one string for `/0/*` and one string for `/1/*`, we can let the Coordinator constructs a generic string at `/*`, and let the Signers worry about post-XPUB restrictions.